TouchPortal-API/TouchPortalAPI/__init__.py

```diff
@@ -166,54 +164,76 @@ def __getWriteLock(self):
         self.__die(exc=RuntimeError("Send buffer mutex deadlock, cannot continue."))
         return False
 
-    def isActionBeingHeld(self, actionId):
+    def isActionBeingHeld(self, actionId:str):
         return actionId in self.__heldActions
 
-    def createState(self, stateId, description, value):
-        if stateId != None and stateId != "" and description != None and description != "" and value != None and value != "":
+    def createState(self, stateId:str, description:str, value:str):
+        if stateId and description and value != None:
             if stateId not in self.currentStates:
                 self.send({"type": "createState", "id": stateId, "desc": description, "defaultValue": value})
                 self.currentStates[stateId] = value
             else:
                 self.stateUpdate(stateId, value)
 
-    def removeState(self, stateId):
-        if stateId in self.currentStates:
+    def createStateMany(self, states:list):
+        try:
+            for state in states:
+                if isinstance(state, dict):
+                    self.createState(state.get('id', ""), state.get('desc', ""), state.get('value', ""))
+                else:
+                    raise TypeError(f'createStateMany() requires a list of dicts, got {type(state)} instead.')
+        except:
+            raise TypeError(f'createStateMany() requires an iteratable, got {type(states)} instead.')
+
+    def removeState(self, stateId:str, validateExists = True):
```

Comment:
I would personally default `validateExists` to `False` or even remove that check entirely... but if anyone is depending on the current behavior then it would of course break.  The API user could already check for themselves if the state exists before trying to remove it, which seems a lot simpler than trapping for an error.